pkg/kubelet/stats/cadvisor_stats_provider.go

```diff
@@ -192,6 +202,19 @@ func isPodManagedContainer(cinfo *cadvisorapiv2.ContainerInfo) bool {
 	return managed
 }
 
+// getcadvisorPodInfoFromPodUID returns a pod cgroup information by matching the podUID with its CgroupName identifier base name
+func getcadvisorPodInfoFromPodUID(podUID types.UID, infos map[string]cadvisorapiv2.ContainerInfo) *cadvisorapiv2.ContainerInfo {
+	for key, info := range infos {
```

Comment:
this is hacky, we should be able to not iterate over every container on the host.

why cant we get the cgroup for a given pod and just do a direct map lookup?

Reply:
We looked into that approach, e.g., GetPodContainerName https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/cm/pod_container_manager_linux.go#L95 can return the pod cgroup name, but it requires pod spec to figure out the qos of a pod. Here cadvisor does not have a pod object

Reply:
i guess this is similar to what we had to do here:
https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/cm/pod_container_manager_linux.go#L200

in that code block we iterate over the literal cgroupfs, convert the full path to our internal canonical name, get the base path, and check for match.

maybe we can find a way to share that code into a common utility so that we can take this knowledge out of this method and keep it localized in cm package in a follow-on?

Reply:
Yes, I will work on that as a follow up PR. Thanks!